Mathlib/CategoryTheory/Abelian/ProjectiveDimension.lean

```diff
@@ -0,0 +1,178 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import Mathlib.Algebra.Homology.DerivedCategory.Ext.ExactSequences
+
+/-!
+# Projective dimension
+
+In an abelian category `C`, we shall say that `X : C` has projective dimension `< n`
+if all `Ext X Y i` vanish when `n ≤ i`.
+
+-/
+
+universe w v u
+
+namespace CategoryTheory
+
+open Abelian Limits ZeroObject
+
+variable {C : Type u} [Category.{v} C] [Abelian C]
+
+/-- An object `X` in an abelian category has projective dimension `< n` if
+all `Ext X Y i` vanish when `n ≤ i`.
+(Do not use the `subsingleton'` field directly. Use the constructor
+`HasProjectiveDimensionLT.mk`, and the lemmas `hasProjectiveDimensionLT_iff` and
+`Ext.eq_zero_of_hasProjectiveDimensionLT`.) -/
+class HasProjectiveDimensionLT (X : C) (n : ℕ) : Prop where mk' ::
```

Comment:
What are the pros and cons of `HasProjectiveDimensionLT` vs `HasProjectiveDimensionLE`. In other words, swapping the role of `\le` and `<`.

I would say $\mathbb{Z}$ has projective dimension $\le 1$, but of course it is also $< 2$.

Does it matter which one we use in practice? I guess `n = 0` is a boundary case. If that is the reason, please mention it in the docstring.

Reply:
Thanks for the suggestion. I have added a type class `HasProjectiveDimensionLE X n` as an abbreviation for `HasProjectiveDimensionLT X (n + 1)`, and it seems Lean is able to see through it.